cpp/ql/src/semmle/code/cpp/ir/implementation/unaliased_ssa/internal/SSAConstruction.qll

```diff
@@ -10,54 +16,35 @@ import Cached
 
 cached
 private module Cached {
-  private IRBlock getNewBlock(OldBlock oldBlock) {
-    result.getFirstInstruction() = getNewInstruction(oldBlock.getFirstInstruction())
+  class TStageInstruction =
+    TRawInstruction or TPhiInstruction or TChiInstruction or TUnreachedInstruction;
+
+  private TRawInstruction rawInstruction(
+    IRFunctionBase irFunc, Opcode opcode, Language::AST ast, Language::LanguageType resultType
+  ) {
+    result = TRawInstruction(irFunc, opcode, ast, resultType, _, _) and
+    result instanceof OldInstruction
   }
 
   cached
-  predicate functionHasIR(Language::Function func) {
-    exists(OldIR::IRFunction irFunc | irFunc.getFunction() = func)
+  predicate hasInstruction(TStageInstruction instr) {
+    instr instanceof TRawInstruction and instr instanceof OldInstruction
+    or
+    not instr instanceof TRawInstruction
```

Comment:
Isn't this just an implication in disguise? i.e., this should be equivalent to:

```codeql
instr instanceof TRawInstruction implies instr instanceof OldInstruction
```

which reads easier (at least to me).

Reply:
Indeed it is. Fixed.